### ISSUE TITLE
-X overloads in File::stat return a list

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#12008]
Migrated from [rt.perl.org#111864](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=111864) (status was 'resolved')

Searchable as RT111864$

**Comment by p5pRT** [comment]
### From @tonycoz

#### Created by @tonycoz



The \-X overloads in blead now return a list containing the File&#8203;::stat
object and the expected result\.

eg\.

5\.12\.3&#8203;:
$ perl \-MFile&#8203;::stat \-le '$s =stat\(\*STDOUT\); print join\("\,"\, \-w $s\)'
1

blead&#8203;:
$ \./perl \-Ilib \-MFile&#8203;::stat \-le '$s =stat\(\*STDOUT\); print join\("\,"\, \-w $s\)'
File&#8203;::stat=ARRAY\(0x159cf38\)\,1

Bisect with&#8203;:

perl \.\./bisect\.pl \-\-start v5\.10\.0 \-e 'use File&#8203;::stat; $s = stat\(STDIN\); @&#8203;foo = \-r $s; @&#8203;foo == 1 or die'

Result

8db8f6b697e6f705eda3222828417099787adba4 is the first bad commit
commit 8db8f6b697e6f705eda3222828417099787adba4
Author&#8203;: Father Chrysostomos \<sprout@&#8203;cpan\.org>
Date&#8203;:   Mon Jan 23 23&#8203;:36&#8203;:29 2012 \-0800

&nbsp;    \[perl \#77388\] Make stacked \-t work

&nbsp;    Up till now\, \-t was popping too much off the stack when stacked with
&nbsp;    other filetest operators\.

&nbsp;    Since the special use of \_ doesn’t apply to \-t\, we cannot simply have
&nbsp;    it use \_ when stacked\, but instead we pass the argument down from the
\.\.\.

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=low
    module=File::stat

Site configuration information for perl 5.15.9:

Configured by tony at Tue Mar 20 19:11:39 EST 2012.

Summary of my perl5 (revision 5 version 15 subversion 9) configuration:
  Commit id: f6067adc61108c3398de698bb0294d95f09b55ef
  Platform:
    osname=linux, osvers=2.6.32-5-amd64, archname=x86_64-linux
    uname='linux mars 2.6.32-5-amd64 #1 smp mon jan 16 16:22:28 utc 2012 x86_64 gnulinux '
    config_args='-des -Dusedevel'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2',
    cppflags='-fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.4.5', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -fstack-protector -L/usr/local/lib'
    libpth=/usr/local/lib /lib/../lib /usr/lib/../lib /lib /usr/lib /lib64 /usr/lib64
    libs=-lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lc -lgdbm_compat
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lc
    libc=/lib/libc-2.11.3.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.11.3'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector'

Locally applied patches:
    


@INC for perl 5.15.9:
    lib
    /usr/local/lib/perl5/site_perl/5.15.9/x86_64-linux
    /usr/local/lib/perl5/site_perl/5.15.9
    /usr/local/lib/perl5/5.15.9/x86_64-linux
    /usr/local/lib/perl5/5.15.9
    .


Environment for perl 5.15.9:
    HOME=/home/tony
    LANG=en_AU.UTF-8
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/tony/bin:/usr/local/bin:/usr/bin:/bin:/usr/games
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @tonycoz

> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
> 
> The \-X overloads in blead now return a list containing the File&#8203;::stat
> object and the expected result\.
> 
> eg\.
> 
> 5\.12\.3&#8203;:
> $ perl \-MFile&#8203;::stat \-le '$s =stat\(\*STDOUT\); print join\("\,"\, \-w $s\)'
> 1
> 
> blead&#8203;:
> $ \./perl \-Ilib \-MFile&#8203;::stat \-le '$s =stat\(\*STDOUT\); print join\("\,"\, \-w $s\)'
> File&#8203;::stat=ARRAY\(0x159cf38\)\,1
> 
> Bisect with&#8203;:
> 
> perl \.\./bisect\.pl \-\-start v5\.10\.0 \-e 'use File&#8203;::stat; $s = stat\(STDIN\); @&#8203;foo = \-r $s; @&#8203;foo == 1 or die'
> 
> Result
> 
> 8db8f6b697e6f705eda3222828417099787adba4 is the first bad commit
> commit 8db8f6b697e6f705eda3222828417099787adba4
> Author&#8203;: Father Chrysostomos \<sprout@&#8203;cpan\.org>
> Date&#8203;:   Mon Jan 23 23&#8203;:36&#8203;:29 2012 \-0800
> 
>     \[perl \#77388\] Make stacked \-t work
> 
>     Up till now\, \-t was popping too much off the stack when stacked with
>     other filetest operators\.
> 
>     Since the special use of \_ doesn’t apply to \-t\, we cannot simply have
>     it use \_ when stacked\, but instead we pass the argument down from the
> \.\.\.

This is a regression against 5\.14&#8203;:

~/perl/5\.14\.2\-debug/bin/perl \-MFile&#8203;::stat \-le '$s =stat\(\*STDOUT\); print join\("\,"\, \-w $s\)'
1

Given the commit that caused the change in behaviour\, other \-X
overrides \(are there any?\) may have similar problems\.

Tony

**Comment by p5pRT** [comment]
### From @cpansprout

On Tue Mar 20 15&#8203;:45&#8203;:46 2012\, tonyc wrote&#8203;:
> > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> > \[Please describe your issue here\]
> >
> > The \-X overloads in blead now return a list containing the
> File&#8203;::stat
> > object and the expected result\.
> >
> > eg\.
> >
> > 5\.12\.3&#8203;:
> > $ perl \-MFile&#8203;::stat \-le '$s =stat\(\*STDOUT\); print join\("\,"\, \-w $s\)'
> > 1
> >
> > blead&#8203;:
> > $ \./perl \-Ilib \-MFile&#8203;::stat \-le '$s =stat\(\*STDOUT\); print join\("\,"\,
> \-w $s\)'
> > File&#8203;::stat=ARRAY\(0x159cf38\)\,1
> >
> > Bisect with&#8203;:
> >
> > perl \.\./bisect\.pl \-\-start v5\.10\.0 \-e 'use File&#8203;::stat; $s =
> stat\(STDIN\); @&#8203;foo = \-r $s; @&#8203;foo == 1 or die'
> >
> > Result
> >
> > 8db8f6b697e6f705eda3222828417099787adba4 is the first bad commit
> > commit 8db8f6b697e6f705eda3222828417099787adba4
> > Author&#8203;: Father Chrysostomos \<sprout@&#8203;cpan\.org>
> > Date&#8203;:   Mon Jan 23 23&#8203;:36&#8203;:29 2012 \-0800
> >
> >     \[perl \#77388\] Make stacked \-t work
> >
> >     Up till now\, \-t was popping too much off the stack when stacked
> with
> >     other filetest operators\.
> >
> >     Since the special use of \_ doesn’t apply to \-t\, we cannot simply
> have
> >     it use \_ when stacked\, but instead we pass the argument down
> from the
> > \.\.\.
> 
> This is a regression against 5\.14&#8203;:
> 
> ~/perl/5\.14\.2\-debug/bin/perl \-MFile&#8203;::stat \-le '$s =stat\(\*STDOUT\);
> print join\("\,"\, \-w $s\)'
> 1
> 
> Given the commit that caused the change in behaviour\, other \-X
> overrides \(are there any?\) may have similar problems\.
> 
> Tony

Well\, I clearly screwed up somehow\.  Can this please be a blocker?  \(Do
I even need to ask?\)


\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @tonycoz

On Wed\, Mar 21\, 2012 at 06&#8203;:10&#8203;:54PM \-0700\, Father Chrysostomos via RT wrote&#8203;:
> On Tue Mar 20 15&#8203;:45&#8203;:46 2012\, tonyc wrote&#8203;:
> > This is a regression against 5\.14&#8203;:
> > 
> > ~/perl/5\.14\.2\-debug/bin/perl \-MFile&#8203;::stat \-le '$s =stat\(\*STDOUT\);
> > print join\("\,"\, \-w $s\)'
> > 1
> > 
> > Given the commit that caused the change in behaviour\, other \-X
> > overrides \(are there any?\) may have similar problems\.
> > 
> > Tony
> 
> Well\, I clearly screwed up somehow\.  Can this please be a blocker?  \(Do
> I even need to ask?\)

I think it should be a blocker\.

Tony

**Comment by p5pRT** [comment]
### From @rjbs

\* Tony Cook \<tony@&#8203;develop\-help\.com> \[2012\-03\-21T22&#8203;:56&#8203;:44\]
> > 
> > Well\, I clearly screwed up somehow\.  Can this please be a blocker?  \(Do
> > I even need to ask?\)
> 
> I think it should be a blocker\.

Me\, too\.  It's been marked\.

\-\- 
rjbs

**Comment by p5pRT** [comment]
### From @cpansprout

Fixed in commit d6642e4\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
